www/js/metrics.js

```diff
@@ -1157,5 +1313,13 @@ angular.module('emission.main.metrics',['nvd3', 'emission.services', 'ionic-date
         return ($scope.expandedc)? "expanded-calorie-card" : "small-calorie-card";
   }
 
-  
+  $scope.openRecentTrip = function(tripId, trip_ts) {
+    $rootScope.recentTripID = tripId;
+    $rootScope.recentTripDate = moment(trip_ts * 1000);
+    $rootScope.tripTimelineUpdate = true;
+    $state.go('root.main.diary').then(function() {
```

Comment:
Then, call this `$state.go('root.main.diary')` with stateParams, e.g.
`$state.go('root.main.diary', {'date': moment(trip_ts * 1000).startOf('day')})`  or something like that.
Combined with the change in `list.js`, it will ensure that the timeline for the correct day is loaded.
And then when you get the TIMELINE_DONE event, you can switch to the detail view with stateParams = tripid and everything should work.